blocks/autocompleters/test/index.js

```diff
@@ -40,40 +46,55 @@ describe( 'blockAutocompleter', () => {
 
 	it( 'should prioritize common blocks in options', () => {
 		return blockAutocompleter( {} ).getOptions().then( ( options ) => {
-			// Exclude React element from label for assertion, since we can't
-			// easily test equality.
-			options = options.map( ( option ) => {
-				expect( option.label[ 0 ].type ).toBe( BlockIcon );
+			expect( options ).toMatchObject( [
+				blockTypes[ 'core/foo' ],
+				blockTypes[ 'core/baz' ],
+				blockTypes[ 'core/bar' ],
+			] );
+		} );
+	} );
 
-				return {
-					...option,
-					label: option.label.slice( 1 ),
-				};
-			} );
+	it( 'should render a block option label composed of @wordpress/element Elements and/or strings', () => {
+		expect.hasAssertions();
 
-			expect( options ).toEqual( [
-				{
-					keywords: [ 'keyword', 'foo' ],
-					label: [
-						'foo',
-					],
-					value: 'core/foo',
-				},
-				{
-					keywords: [ 'baz' ],
-					label: [
-						'baz',
-					],
-					value: 'core/baz',
-				},
-				{
-					keywords: [ 'bar' ],
-					label: [
-						'bar',
-					],
-					value: 'core/bar',
-				},
-			] );
+		const completer = blockAutocompleter( { } );
+		const renderedLabels = getBlockTypes().map(
+			blockType => completer.renderOptionLabel( blockType )
```

Comment:
It looks like you can perform all assertions inside this loop.

Reply:
Updated.